mira/dkg/resources/extract_eiffel_ontology.py

```diff
@@ -0,0 +1,433 @@
+        individual_uri = str(res['individual'])
+        compressed_individual = converter.compress(individual_uri)
+        description_str = res['description']
+        if description_str is None:
+            description_str = ''
+        graph_obj.info_list.append(
+            (compressed_individual, individual_uri, str(res['label']),
+             description_str))
+
+    relationship_query = """
+        SELECT ?subject ?predicate ?object ?label
+        WHERE{
+            {?subject ?predicate ?object}. 
+            ?subject rdfs:label ?label 
+            FILTER (isIRI(?object) && (CONTAINS(STR(?predicate), "ecv")))
+        }
+    """
+    relationship_results_dict = graph_obj.graph.query(relationship_query)
+
+    for res in relationship_results_dict:
+        subject_uri = str(res['subject'])
+        compressed_subject = converter.compress(subject_uri)
+        predicate_uri = str(res['predicate'])
+        compressed_predicate = converter.compress(predicate_uri)
+        object_uri = str(res['object'])
+        compressed_object = converter.compress(object_uri)
+        graph_obj.relationship_list.append((str(res['label']),
+                                            compressed_subject,
+                                            subject_uri,
+                                            compressed_predicate,
+                                            predicate_uri,
+                                            compressed_object, object_uri))
+
+    unique_predicate_query = """
+            SELECT DISTINCT ?predicate
+            WHERE {
+              ?subject ?predicate ?object.
+            }
+            """
+
+    unique_predicate_results = graph_obj.graph.query(unique_predicate_query)
+    unique_predicates = list(
+        set(str(row['predicate']) for row in
+            unique_predicate_results.bindings))
+
+    with open(str(RESULTS_DIR) + '/' + graph_obj.name + ' information.csv',
+              'w') as file:
+        csv_out = csv.writer(file)
+        csv_out.writerow(['Individual', 'Individual URI', 'Label',
+                          'Description'])
+        for row in graph_obj.info_list:
+            csv_out.writerow(row)
+
+    with open(str(RESULTS_DIR) + '/' + graph_obj.name + ' relation.csv',
+              'w') as file:
+        csv_out = csv.writer(file)
+        csv_out.writerow(['Label', 'Subject', 'Subject URI', 'Predicate',
+                          'Predicate URI', 'Object', 'Object URI'])
+
+        for row in graph_obj.relationship_list:
+            csv_out.writerow(row)
+
+    with open(str(RESULTS_DIR) + '/' + graph_obj.name +
+              ' unique_predicates.txt', 'w') as file:
+        file.write(
+            f'Number of unique predicates: {len(unique_predicates)}\n\n')
+
+        # Write each unique predicate to the file
+        for predicate in unique_predicates:
+            file.write(f'{predicate}\n')
+
+
+def process_eo(graph_obj, converter):
+    info_query = """
```

Comment:
save all of these queries as top-level constants so as to declutter the functions